include/onnxruntime/core/graph/graph_nodes.h

```diff
@@ -122,7 +122,7 @@ class ValidNodes {
 
    private:
     TIterator current_;
-    const TIterator end_;
+    TIterator end_;
```

Comment:
It would have no effect. Here this is necessary so the iterator becomes copyable and usable for std algorigthms that create a copy of the iterator args

---
In reply to: [267052891](https://github.com/Microsoft/onnxruntime/pull/651#discussion_r267052891) [](ancestors = 267052891)

Reply:
We're taking copies (as per usual iterator semantics), so I'd probably go the other way and make 'current' in the ctor const as well.

Reply:
Agreed -- just wanted to make sure the two are consistent.